src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -984,54 +941,24 @@ subroutine diabatic_ALE_legacy(u, v, h, tv, Hml, fluxes, visc, ADp, CDp, dt, Tim
-      ! Changes T and S via the tridiagonal solver; no change to h
-      do k=1,nz ; do j=js,je ; do i=is,ie
-        ea_t(i,j,k) = ea_s(i,j,k) ; eb_t(i,j,k) = eb_s(i,j,k)
-      enddo ; enddo ; enddo
-      if (CS%tracer_tridiag) then
-        call tracer_vertdiff(hold, ea_t, eb_t, dt, tv%T, G, GV)
-        call tracer_vertdiff(hold, ea_s, eb_s, dt, tv%S, G, GV)
-      else
-        call triDiagTS(G, GV, is, ie, js, je, hold, ea_s, eb_s, tv%T, tv%S)
-      endif
-
-      ! diagnose temperature, salinity, heat, and salt tendencies
-      ! Note: hold here refers to the thicknesses from before the dual-entraintment when using
-      ! the bulk mixed layer scheme. Otherwise in ALE-mode, layer thicknesses will (not?) have changed
-      ! In either case, tendencies should be posted on hold
-      if (CS%diabatic_diff_tendency_diag) then
-        call diagnose_diabatic_diff_tendency(tv, hold, temp_diag, saln_diag, dt, G, GV, US, CS)
-        if (CS%id_diabatic_diff_h > 0) call post_data(CS%id_diabatic_diff_h, hold, CS%diag, alt_h = hold)
-      endif
-    else
-      ! Set ea_t=eb_t based on Kd_heat and ea_s=eb_s based on Kd_salt on interfaces for use in the tri-diagonal solver.
-
-      do j=js,je ; do i=is,ie
-        ea_t(i,j,1) = 0.; ea_s(i,j,1) = 0.
-      enddo ; enddo
-
-      !$OMP parallel do default(shared) private(hval)
-      do k=2,nz ; do j=js,je ; do i=is,ie
-        hval = 1.0 / (h_neglect + 0.5*(h(i,j,k-1) + h(i,j,k)))
-        ea_t(i,j,k) = (GV%Z_to_H**2) * dt * hval * Kd_heat(i,j,k)
-        eb_t(i,j,k-1) = ea_t(i,j,k)
-        ea_s(i,j,k) = (GV%Z_to_H**2) * dt * hval * Kd_salt(i,j,k)
-        eb_s(i,j,k-1) = ea_s(i,j,k)
-      enddo ; enddo ; enddo
-      do j=js,je ; do i=is,ie
-        eb_t(i,j,nz) = 0. ; eb_s(i,j,nz) = 0.
-      enddo ; enddo
-      if (showCallTree) call callTree_waypoint("done setting ea_t,ea_s,eb_t,eb_s from Kd_heat" //&
-         "and Kd_salt (diabatic)")
-
     ! Changes T and S via the tridiagonal solver; no change to h
-      call tracer_vertdiff(h, ea_t, eb_t, dt, tv%T, G, GV)
-      call tracer_vertdiff(h, ea_s, eb_s, dt, tv%S, G, GV)
+    do k=1,nz ; do j=js,je ; do i=is,ie
+      ea_t(i,j,k) = ea_s(i,j,k) ; eb_t(i,j,k) = eb_s(i,j,k)
+    enddo ; enddo ; enddo
+    if (CS%tracer_tridiag) then
+      call tracer_vertdiff(hold, ea_t, eb_t, dt, tv%T, G, GV)
+      call tracer_vertdiff(hold, ea_s, eb_s, dt, tv%S, G, GV)
+    else
+      call triDiagTS(G, GV, is, ie, js, je, hold, ea_s, eb_s, tv%T, tv%S)
+    endif
 
-      ! In ALE-mode, layer thicknesses do not change. Therefore, we can use h below
-      if (CS%diabatic_diff_tendency_diag) &
-        call diagnose_diabatic_diff_tendency(tv, h, temp_diag, saln_diag, dt, G, GV, US, CS)
+    ! diagnose temperature, salinity, heat, and salt tendencies
+    ! Note: hold here refers to the thicknesses from before the dual-entraintment when using
+    ! the bulk mixed layer scheme. Otherwise in ALE-mode, layer thicknesses will (not?) have changed
```

Comment:
Since `eatr(k+1)=ebtr(k)` in ALE mode, this diagnostic should always be 0. Maybe don't register in ALE mode?